tpm.go

```diff
@@ -217,38 +219,39 @@ func (t *TPMConnection) init() error {
 		if err == nil {
 			return ek, nil
 		}
-		if !tpm2.IsResourceUnavailableError(err, ekHandle) {
+		if !tpm2.IsResourceUnavailableError(err, ekHandle) || !secureMode {
 			return nil, err
 		}
 		if ek, err := createTransientEk(t.TPMContext); err == nil {
 			return ek, nil
 		}
 		return nil, err
 	}()
-	if err != nil && len(t.verifiedEkCertChain) > 0 {
+	if err != nil && secureMode {
 		// A lack of EK should be fatal in this context
 		return xerrors.Errorf("cannot obtain context for EK: %w", err)
 	}
 
-	ekIsPersistent := false
```

Comment:
even if we don't want to keep the flag and update it, I think a local function:

```
ekIsPersistent := func() bool {
     return ek != nil && ek.Handle() == ekHandle
}
```

could make the code slightly more readable